chore: upgrade NumPy to 2.0

benglewis opened the pull request:
Fixes #296 
Update NumPy version to 2.0.0 in the repository.

* **pyproject.toml**
  - Change NumPy version from `>=1.26.4,<2.0.0` to `2.0.0`.

* **examples/mensuration_and_polling/requirements.txt**
  - Change NumPy version from `~=1.24.3` to `~=2.0.0`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/HumanSignal/label-studio-sdk/pull/436?shareId=f184388c-79e5-448b-bdbc-ff9797668ee0).

pakelley commented:
I relaxed the pin on this to still allow numpy 1. I know you're out the next couple days, but I'm going to go ahead and get this merged. If there's any issue with that change lmk and we can resolve it when you're back 🙂

pakelley commented:
Ugh, it looks like the `POETRY_VERSION` still isn't being set on this PR (though I do see it properly installing poetry on your PR to your own master branch...)
Mind taking a look at your variables again when you're back in? As soon as we get CI passing this can be (finally) merged in 🙂 
Thank you for your contribution!!

benglewis commented:
@pakelley Unfortunately I don't think that there is anything that I can do about this since it looks like it is a GitHub Actions restriction:
https://github.com/orgs/community/discussions/44322
I guess that you could just maybe add a fallback for the variable in the GitHub Actions workflow, but clearly that isn't an ideal solution 🙊🙈

benglewis commented:
@pakelley I have pulled updates from the upstream `main` and updated the `poetry.lock` file and I have added a fallback value for `POETRY_VERSION` to the PyTest tests. This should mean that this PR should pass this time around 🙏 I think that you just need to re-approve for the tests to re-run

nikitabelonogov commented:
Hi @benglewis I decide to go with slightly different approach for poetry version https://github.com/HumanSignal/label-studio-sdk/commit/68d512f70445203a55e7ba003feca0b0d8ab20e1

merged to your branch

benglewis commented:
@nikitabelonogov Great. Can we merge this now then? 🙊

farioas commented:
@benglewis Sorry for the confusion, we are only here to fix the build problem.

@pakelley This change will be overridden by fern on the next run by:
https://github.com/HumanSignal/label-studio-client-generator/blob/a59e801858ae432d4b2c164629a32b1cae30d0e6/fern/generators.yml#L49

pakelley commented:
Ah, I see, I'll update that in the client generator, thanks @farioas

pakelley commented:
client generator PR [here](https://github.com/HumanSignal/label-studio-client-generator/pull/69), look good to you @farioas and @nikitabelonogov ?

micaelakaplan commented:
Hi @benglewis ! Could you email me regarding this PR? micaela@humansignal.com thanks!
